### ISSUE TITLE
CONTAINERS_CONF is not propagated to conmon

### ISSUE THREAD

**Issue by dgibson** [containers/podman#12917]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

The `CONTAINERS_CONF` environment variable can be used to locally override the path to find the `containers.conf` configuration file.  This ise useful for testing various scenarios.  However, to work properly, this override needs to apply to all podman invocations through the container lifecycle.  At present the `CONTAINERS_CONF` variable is not propagated to `conmon`, which in particular means that it is not applied to the `podman container cleanup` call invoked by `conmon` when the container terminates.

**Steps to reproduce the issue:**

1.  Create a local `containers.conf` with an unusual element.  For example this configuration uses an unusual runtime name, even though it's just an alternative name for crun:
```
[engine]
runtime = "weirdname"
[engine.runtimes]
weirdname = ["/bin/crun"]
```
2. Run this command (here, `testimage` is the sample container image from the podman system tests)
```
$ CONTAINERS_CONF=~/tmp/containers.conf podman run -d --rm testimage true
8783ebedc0c1399afefe62de16dd76f43775a520b90c931cc2dda3b26600fafc
```
3. Check running containers:
```
$ CONTAINERS_CONF=~/tmp/containers.conf podman ps -a
```

**Describe the results you received:**

Because `CONTAINERS_CONF` is not propagated to the `podman container cleanup` call, that call fails, which means (amongst other things) that the container is not removed after completion despite the `--rm` flag:

```
CONTAINER ID  IMAGE                              COMMAND     CREATED         STATUS                     PORTS       NAMES
8783ebedc0c1  quay.io/libpod/testimage:20210610  true        15 seconds ago  Exited (0) 15 seconds ago              goofy_ishizaka
```

**Describe the results you expected:**

`podman container cleanup` to be called with the correct configuration, therefore removing the container.

**Output of `podman version`:**

```
podman version 3.4.4
```

The problem also occurs with a locally built podman, from commit `b59c30f729401214131dd1378a49efa80f134717`.

**Output of `podman info --debug`:**

```
$ podman info --debug
ERRO[0000] OCI Runtime weirdname is in use by a container, but is not available (not in configuration file or not installed) 
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.30-2.fc35.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.30, commit: '
  cpus: 8
  distribution:
    distribution: fedora
    version: "35"
  eventLogger: journald
  hostname: yekko
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.15.14-200.fc35.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 17040199680
  memTotal: 33424232448
  ociRuntime:
    name: crun
    package: crun-1.4-1.fc35.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4
      commit: 3daded072ef008ef0840e8eccb0b52a7efbd165d
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /bin/slirp4netns
    package: slirp4netns-1.1.12-2.fc35.x86_64
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 14h 6m 8.6s (Approximately 0.58 days)
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/dwg/.config/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 0
    stopped: 1
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/dwg/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 1
  runRoot: /run/user/1000/containers
  volumePath: /home/dwg/.local/share/containers/storage/volumes
version:
  APIVersion: 3.4.4
  Built: 1638999907
  BuiltTime: Thu Dec  9 08:45:07 2021
  GitCommit: ""
  GoVersion: go1.16.8
  OsArch: linux/amd64
  Version: 3.4.4
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.4.4-1.fc35.x86_64
```

**Comment by dgibson** [comment]
I have a patch for this problem that I'm polishing up and will submit soon.

**Comment by dgibson** [comment]
Addressed in #12918
